vm/src/stdlib/os.rs

```diff
@@ -2539,21 +2539,52 @@ mod posix {
 
     #[pyfunction]
     fn chmod(
-        path: PyPathLike,
+        path: PathOrFd,
         dir_fd: DirFd<0>,
         mode: u32,
         follow_symlinks: FollowSymlinks,
         vm: &VirtualMachine,
     ) -> PyResult<()> {
         let [] = dir_fd.0;
-        let body = move || {
-            use std::os::unix::fs::PermissionsExt;
-            let meta = fs_metadata(&path, follow_symlinks.0)?;
-            let mut permissions = meta.permissions();
-            permissions.set_mode(mode);
-            fs::set_permissions(&path, permissions)
-        };
-        body().map_err(|err| err.into_pyexception(vm))
+        match path {
+            PathOrFd::Path(path) => {
+                let body = move || {
+                    use std::os::unix::fs::PermissionsExt;
+                    let meta = fs_metadata(&path, follow_symlinks.0)?;
+                    let mut permissions = meta.permissions();
+                    permissions.set_mode(mode);
+                    fs::set_permissions(&path, permissions)
+                };
+                body().map_err(|err| err.into_pyexception(vm))
+            }
+            PathOrFd::Fd(fd) => nix::sys::stat::fchmod(
+                fd,
+                nix::sys::stat::Mode::from_bits(mode as libc::mode_t).unwrap(),
+            )
+            .map_err(|err| err.into_pyexception(vm)),
```

Comment:
is this same as calling `fchmod`?

Reply:
Yes. In https://docs.python.org/3/library/os.html#os.fchmod , 

> As of Python 3.3, this is equivalent to os.chmod(fd, mode).

Reply:
Then I expect this is just calling fchmod.

Reply:
I also want lchmod and chmod share their parts if they do same function.

Reply:
OK. I will make them integrate with chmod.